app/uk/gov/hmrc/individualsemploymentsapi/service/v2/CacheServiceV2.scala

```diff
@@ -26,17 +29,39 @@ import scala.concurrent.{ExecutionContext, Future}
 class CacheServiceV2 @Inject()(cachingClient: ShortLivedCacheV2, conf: CacheConfigurationV2)(
   implicit ec: ExecutionContext) {
 
-  def get[T: Format](cacheId: String, functionToCache: => Future[T])(implicit hc: HeaderCarrier): Future[T] =
+  def get[T: Format](cacheId: CacheIdBase, functionToCache: => Future[T])(implicit hc: HeaderCarrier): Future[T] =
     if (conf.cacheEnabled) {
-      cachingClient.fetchAndGetEntry[T](cacheId, conf.key) flatMap {
+      cachingClient.fetchAndGetEntry[T](cacheId.id, conf.key) flatMap {
         case Some(value) => Future.successful(value)
         case None =>
           functionToCache map { res =>
-            cachingClient.cache(cacheId, conf.key, res)
+            cachingClient.cache(cacheId.id, conf.key, res)
             res
           }
       }
     } else {
       functionToCache
     }
 }
+
+// Cache ID implementations
+// This can then be concatenated for multiple scopes.
+// Example;
+// read:scope-1 =  [A, B, C]
+// read:scope-2 = [D, E, F]
+// The cache key (if two scopes alone) would be;
+// `id + from + to +  [A, B, C, D, E, F]` Or formatted to `id-from-to-ABCDEF`
+// The `fields` param is obtained with scopeService.getValidFieldsForCacheKey(scopes: List[String])
+
+trait CacheIdBase {
+  val id: String
+
+  override def toString: String = id
+}
+
+case class CacheIdV2(matchId: UUID, interval: Interval, fields: String) extends CacheIdBase {
```

Comment:
I don't think there's a CacheIdV1 - can we change this to CacheId? I'm not keen on version numbers in class names